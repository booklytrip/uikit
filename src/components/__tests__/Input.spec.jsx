import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Input } from '../Input';
import Icon from '../Icon';

describe('Input', () => {
    it('renders without crashing', () => {
        mount(<Input />);
    });
    
    it('should call onChange event if changed', () => {
        const spy = jest.fn();
        const input = mount(<Input onChange={ spy } />);
        input.find('input').simulate('change');
        
        expect(spy).toHaveBeenCalled();
    });
    
    describe('Snapshots', () => {
        it('renders showing that data contains error', () => {
            const rendered = renderer.create(<Input error />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with icon on left position', () => {
            const rendered = renderer.create(
                <Input
                    icon={ <Icon name="user" /> }
                    labelPosition="left"
                />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders with icon on right position', () => {
            const rendered = renderer.create(
                <Input
                    icon={ <Icon name="user" /> }
                    labelPosition="right"
                />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with different size', () => {
            const rendered = renderer.create(<Input size="small" />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
