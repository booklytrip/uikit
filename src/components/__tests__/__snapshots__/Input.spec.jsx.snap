exports[`Input Snapshots renders showing that data contains error 1`] = `
<div
  className="ui input error">
  <input />
</div>
`;

exports[`Input Snapshots renders with different size 1`] = `
<div
  className="ui input small">
  <input />
</div>
`;

exports[`Input Snapshots renders with icon on left position 1`] = `
<div
  className="ui input left labeled">
  <input />
  <i
    className="icon user"
    onClick={undefined}
    style={undefined} />
</div>
`;

exports[`Input Snapshots renders with icon on right position 1`] = `
<div
  className="ui input right labeled">
  <input />
  <i
    className="icon user"
    onClick={undefined}
    style={undefined} />
</div>
`;
